cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -110,3 +120,31 @@ def test_edit_volume_with_type(volume_with_type, appliance):
 def test_delete_volume_with_type(volume_with_type):
     volume_with_type.delete()
     assert not volume_with_type.exists
+
+
+@pytest.mark.regression
+def test_volume_attach_detach_instance(volume, new_instance, appliance):
+    initial_instance_count = volume.instance_count
+    volume.attach_instance(new_instance.name)
+    view = appliance.browser.create_view(navigator.get_class(volume, 'Details').VIEW)
+    view.flash.assert_success_message(
+        'Attaching Cloud Volume "{name}" to {instance} finished'.format(
+            name=volume.name, instance=new_instance.name))
+
+    @wait_for_decorator(delay=10, timeout=300,
+                        message="Waiting for volume to be attached to instance")
+    def volume_attached_to_instance():
+        new_instance.refresh_relationships()
+        return volume.instance_count > initial_instance_count
```

Comment:
@jadhaj Have you thought about ```return volume.instance_count == initial_instance_count + 1``` here? This test should be the only one adding this specific volume to an instance. This would make it a more specific check.

Reply:
Actually i thought about  return volume.instance_count ==1 and  return volume.instance_count ==0 without using initial_instance_count var

Reply:
@jadhaj I think it be better to be specific here, so I going to mark this as a **Required** change.